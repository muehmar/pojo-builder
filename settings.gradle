rootProject.name = "pojo-builder"

include 'core'
include 'core-annotations'
include 'example'
include 'example-java17'

dependencyResolutionManagement {
    versionCatalogs {
        libs {
            version('pojobuilder', '1.6.0')
            library('pojobuilder-core', 'io.github.muehmar','pojo-builder').versionRef('pojobuilder')
            library('pojobuilder-annotations', 'io.github.muehmar','pojo-builder-annotations').versionRef('pojobuilder')

            library('codeGenerator', 'io.github.muehmar:code-generator:0.22.0')

            version('junit', '5.10.0')
            library('junit-api', 'org.junit.jupiter', 'junit-jupiter-api').versionRef('junit')
            library('junit-engine', 'org.junit.jupiter', 'junit-jupiter-engine').versionRef('junit')
            library('junit-params', 'org.junit.jupiter', 'junit-jupiter-params').versionRef('junit')

            version('assertj', '3.26.3')
            library('assertj', 'org.assertj', 'assertj-core').versionRef('assertj')

            bundle('junit', ['junit-api', 'junit-engine', 'junit-params', 'assertj'])

            library('jakarta-annotationApi', 'jakarta.annotation:jakarta.annotation-api:2.1.1')

            library('google-autoservice', 'com.google.auto.service:auto-service:1.1.1')
            library('joor', 'org.jooq:joor-java-8:0.9.15')

            library('lombok', 'org.projectlombok:lombok:1.18.32')

            library('snapshotTesting','io.github.origin-energy:java-snapshot-testing-junit5:4.0.8')

            library('slf4j', 'org.slf4j:slf4j-api:2.0.7')

            plugin('release', 'net.researchgate.release').version('2.8.1')
            plugin('versions', 'com.github.ben-manes.versions').version('0.51.0')
        }
    }
}