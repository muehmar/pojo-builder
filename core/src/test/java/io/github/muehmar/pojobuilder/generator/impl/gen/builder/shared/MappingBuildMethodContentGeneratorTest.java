package io.github.muehmar.pojobuilder.generator.impl.gen.builder.shared;

import static io.github.muehmar.codegenerator.writer.Writer.javaWriter;
import static io.github.muehmar.pojobuilder.generator.impl.gen.builder.shared.MappingBuildMethodContentGenerator.mappingBuildMethodContentGenerator;
import static io.github.muehmar.pojobuilder.snapshottesting.SnapshotUtil.writerSnapshot;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import ch.bluecare.commons.data.PList;
import io.github.muehmar.codegenerator.Generator;
import io.github.muehmar.codegenerator.writer.Writer;
import io.github.muehmar.pojobuilder.generator.Pojos;
import io.github.muehmar.pojobuilder.generator.model.BuildMethod;
import io.github.muehmar.pojobuilder.generator.model.Name;
import io.github.muehmar.pojobuilder.generator.model.Pojo;
import io.github.muehmar.pojobuilder.generator.model.settings.PojoSettings;
import io.github.muehmar.pojobuilder.generator.model.type.Types;
import io.github.muehmar.pojobuilder.snapshottesting.SnapshotTest;
import java.util.Optional;
import org.junit.jupiter.api.Test;

@SnapshotTest
class MappingBuildMethodContentGeneratorTest {
  private Expect expect;

  @Test
  @SnapshotName("sample")
  void mappingBuildMethodContentGenerator_when_calledWithSample_then_correctOutput() {
    final Generator<Pojo, PojoSettings> generator = mappingBuildMethodContentGenerator();
    final String output =
        generator.generate(Pojos.sample(), PojoSettings.defaultSettings(), javaWriter()).asString();

    expect.toMatchSnapshot(output);
  }

  @Test
  @SnapshotName("genericSample")
  void mappingBuildMethodContentGenerator_when_calledWithGenericSample_then_correctOutput() {
    final Generator<Pojo, PojoSettings> generator = mappingBuildMethodContentGenerator();
    final String output =
        generator
            .generate(Pojos.genericSample(), PojoSettings.defaultSettings(), javaWriter())
            .asString();

    expect.toMatchSnapshot(output);
  }

  @Test
  @SnapshotName("genericSampleAndBuildMethod")
  void
      mappingBuildMethodContentGenerator_when_calledWithGenericSampleAndBuildMethod_then_correctOutput() {
    final Generator<Pojo, PojoSettings> generator = mappingBuildMethodContentGenerator();
    final io.github.muehmar.pojobuilder.generator.model.BuildMethod buildMethod =
        new BuildMethod(Name.fromString("customBuildMethod"), Types.string(), PList.empty());
    final Pojo pojo = Pojos.genericSample().withBuildMethod(Optional.of(buildMethod));
    final String output =
        generator.generate(pojo, PojoSettings.defaultSettings(), javaWriter()).asString();

    expect.toMatchSnapshot(output);
  }

  @Test
  @SnapshotName("sampleAndConstructorWithOptionalArgument")
  void
      mappingBuildMethodContentGenerator_when_generatorUsedWithSamplePojoAndConstructorWithOptionalArgument_then_correctOutput() {
    final Generator<Pojo, PojoSettings> generator = mappingBuildMethodContentGenerator();
    final Writer writer =
        generator.generate(
            Pojos.sampleWithConstructorWithOptionalArgument(),
            PojoSettings.defaultSettings(),
            javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }

  @Test
  @SnapshotName("sampleWithFactoryMethod")
  void
      mappingBuildMethodContentGenerator_when_sampleWithFactoryMethod_then_correctOutputGenerator() {
    final Generator<Pojo, PojoSettings> generator = mappingBuildMethodContentGenerator();
    final Writer writer =
        generator.generate(
            Pojos.factoryMethodSample(), PojoSettings.defaultSettings(), javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }
}
