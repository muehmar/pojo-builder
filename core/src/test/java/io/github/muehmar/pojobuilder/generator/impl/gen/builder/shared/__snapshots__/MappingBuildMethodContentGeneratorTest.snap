genericSample=[
final Customer<T, S> instance =
    new Customer<T, S>(id, data, additionalData);
return instance;
]


genericSampleAndBuildMethod=[
final Customer<T, S> instance =
    new Customer<T, S>(id, data, additionalData);
return Customer.customBuildMethod(instance);
]


sample=[
final Customer instance =
    new Customer(id, username, nickname);
return instance;
]


sampleAndConstructorWithOptionalArgument=[
java.util.Optional

final Customer instance =
    new Customer(id, username, Optional.ofNullable(nickname));
return instance;
]


sampleWithFactoryMethod=[
java.util.Optional

return Pojos.Customer.create(id, username, Optional.ofNullable(nickname));
]